jetbrains-rider/src/software/aws/toolkits/jetbrains/services/clouddebug/DotNetStartupCommand.kt

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.clouddebug
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.jetbrains.rd.framework.RdTaskResult
+import com.jetbrains.rdclient.protocol.IPermittedModalities
+import com.jetbrains.rider.model.AwsProjectOutputRequest
+import com.jetbrains.rider.model.awsProjectModel
+import com.jetbrains.rider.projectView.solution
+import com.jetbrains.rider.util.idea.lifetime
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.core.utils.warn
+import software.aws.toolkits.jetbrains.services.ecs.execution.ArtifactMapping
+import java.io.File
+
+class DotNetStartupCommand : CloudDebugStartupCommand(CloudDebuggingPlatform.DOTNET) {
+
+    companion object {
+        private val logger = getLogger<DotNetStartupCommand>()
+        private const val STARTUP_COMMAND_HINT_TEXT = "Example: dotnet /path/to/assembly.dll"
+    }
+
+    override val isStartCommandAutoGenerateSupported: Boolean
+        get() = true
+
+    override fun updateStartupCommand(
+        project: Project,
+        originalCommand: String,
+        artifact: ArtifactMapping,
+        onCommandGet: (String) -> Unit
+    ) {
+        val model = project.solution.awsProjectModel
+
+        val localPath = artifact.localPath
+        val remotePath = artifact.remotePath
+        if (localPath == null || remotePath == null) {
+            logger.warn { "Local or remote path is empty in an artifacts mapping. Return the original startup command: '$originalCommand'." }
+            onCommandGet(originalCommand)
+            return
+        }
+
+        val localFile = File(localPath)
+        val localBaseFile =
+            if (localFile.isFile) localFile.parentFile
+            else localFile
+
+        val remoteFile = File(remotePath)
+        val remoteBaseFile =
+            if (remoteFile.isFile) {
+                logger.info { "Remote path is set file in Artifacts mapping table: '$remotePath'. Take base path." }
+                remoteFile.parentFile
+            } else
+                remoteFile
+
+        // To get a project output we make a protocol call that is executed on AWT thread using run configuration dialog modality.
+        // Then protocol queue is waiting for return to AWT to be able to pump protocol queue and get a response back.
+        // We should allow pumping protocol queue with current run configuration modality to get a response back.
+        // Note: This is better to allow pumping for a specific component using [PermittedModalitiesImpl#allowPumpProtocolForComponent] API,
+        //       but for a particular case we can use a call that use current modality since we are in the same run config context.
+        IPermittedModalities.getInstance().allowPumpProtocolUnderCurrentModality()
```

Comment:
Do we need to undo anything after call is made?

Reply:
No, everything will be auto-disposed on modality change.